### PR TITLE
feat: add simple code wizard

### LINKED ISSUE #8: Add simple code wizard
As a suer of the `open-scd` component, I want to have at least the code wizard listening for incoming wizard events. This code wizard should allow to change any XML code.

### PR DESCRIPTION
**Pull request by JakobVogelsang**
closes #8

### CODE REVIEW

**Review by trusz — COMMENTED** [review]
I think I would somehow make sure that there is only one shadow backdrop. the multiple shadows on top of each other make the screen a bit noisy :)

File: open-scd.ts [hunk 1]
```diff
@@ -374,6 +383,28 @@ export class OpenSCD extends LitElement {
       .filter(p => p !== undefined);
   }
 
+  /** FIFO queue of [[`Wizard`]]s to display. */
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply open-scd.ts:386]: You are right with the key `subwizard` set to true, which as default is `false`. Per default and in most used case we are seeing a FIFO queue

File: open-scd.ts [hunk 2]
```diff
@@ -399,6 +430,27 @@ export class OpenSCD extends LitElement {
     this.addEventListener('oscd-edit-v2', event =>
       this.handleEditEventV2(event)
     );
+
+    this.addEventListener('oscd-edit-wizard-request', event =>
+      this.onWizard(event as EditWizardEvent)
+    );
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply open-scd.ts:436]: As this is the core component and as it never really get replaced, I would not worry about that. The reason we do register those in the constructor I readability. I think most will not see those intuitively in the template.

[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply open-scd.ts:436]: Just for interest: when I have a custom web component loaded with event listeners added in the constructed and remove the component from the DOM would the event listener be removed as well. This is what I would have assumed, or how would the memory leak be possible?

File: open-scd.ts [hunk 3]
```diff
@@ -374,6 +383,28 @@ export class OpenSCD extends LitElement {
...
+  @state()
+  workflow: WizardRequest[] = [];
+
+  @query('.wizard.code') codeWizard?: CodeWizard;
+
+  private closeWizard(we: CloseWizardEvent): void {
+    const wizard = we.detail;
+
+    this.workflow.splice(this.workflow.indexOf(wizard), 1);
+    this.requestUpdate();
```
[the remark this answers is not in the document]
Comment (JakobVogelsang) [reply open-scd.ts:396]: Not really, thank you.

**Pull request merged**